### ISSUE TITLE
Use BatchLoader in GraphQL logic to compose In clause for SQL query

### ISSUE THREAD

**Issue by landonreed** [conveyal/gtfs-lib#125]
Currently, nested objects in a GraphQL request are fetched in separate SQL queries. For example, if requesting the trips for a pattern with nested stop_times, one query will be made for the n trips and then n SQL queries will be made to fetch the stop_times.

We should use a [BatchLoader](https://github.com/graphql-java/graphql-java/blob/master/docs/batching.rst) to make this more efficient.

cc @abyrd.

**Comment by abyrd** [comment]
Oh nice, I always cringed a little at cascading hundreds of SQL queries, it's good to know there's a more efficient way to handle that.

**Comment by landonreed** [comment]
Yes, I'd like to fast track this because the way we're doing it now is super inefficient.  Every time I see those redundant SQL queries stack up in the logs I also cringe.

**Comment by evansiroky** [comment]
I'm not sure BatchLoader would help in this case.  The BatchLoader works well for when the same entity is repeatedly being requested as a result of being seen in different levels of the graph.  However, another tool may exist that can do SQL joins.  I'll try to look into that.

**Comment by abyrd** [comment]
If I'm not mistaken the device we're talking about is called a DataLoader rather than a BatchLoader (https://github.com/graphql-java/java-dataloader). I changed the name of the issue, but please correct me if I'm wrong. 

@evansiroky I think the cacheing is just one of two optimizations, the other of which is batching requests. Most of the articles I've looked at on the subject mention batching requests together.

https://spin.atomicobject.com/2017/05/15/optimize-graphql-queries/
https://medium.com/@gajus/using-dataloader-to-batch-requests-c345f4b23433
http://www.petecorey.com/blog/2017/08/14/batching-graphql-queries-with-dataloader/

**Comment by landonreed** [comment]
#144 begins to address this. However, the original intent of this ticket was to reduce the number of SQL queries for a GraphQL query that has nested entities (e.g., _n_ trips with nested stop_times) from _n_ individual queries for the nested entity to one query that has an "in clause" containing all of the parent join values. We could then add logic to the DataLoader that is able to filter out the results for each JdbcQuery.
